### PR TITLE
feat: versionware; introduce version-based routing

### PR DESCRIPTION
**Pull request by cmars**
Add a multiplexing handler that dispatches requests to handlers based on
a version query parameter.

### CODE REVIEW

**Comment by cmars** [comment]
@talgendler @jjmschofield you might find this useful

**Comment by cmars** [comment]
@genslein I'm iterating on some feedback there. Starting this as an experiment, a possible lightweight approach to a version dispatch that might also support edge proxy use cases.

File: versionware/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"net/http/httptest"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+
+	"github.com/snyk/vervet"
+	"github.com/snyk/vervet/versionware"
+)
+
+func ExampleHandler() {
```
Comment (jcsackett) [inline versionware/handler_test.go:16]: Where does this get used? I don't see it called in this test--looks like you directly implemented something similar in the tests below. Am I missing something?

**Review by jcsackett — APPROVED** [review]
LGTM, though I'm curious about the one func called out in my comment. I assume I'm missing something :)

File: versionware/handler_test.go (hunk 1 above)
Comment (cmars) [reply versionware/handler_test.go:16]: It gets run like a `Test*` function, but it also [shows up in godocs when you name them like this](https://go.dev/blog/examples). It'll be a nice way to play with the code.

**Pull request merged**